brave/brave-ios
# [v1.58] Fix #7818, #7807: Mojo Origin Cleanup, AccountId for Transaction refactor, and BraveCore v1.58.77 bump

## Summary of Changes

This pull request fixes #7818, fixes #7807

## Submitter Checklist:


## Test Plan:
1. Verify origin (and Favicon when displayed on that view) displays as expected for Get Encryption Public Key request, Decrypt request, Switch Chain request.
2. Verify Get Encryption Public Key request, Decrypt request, Switch Chain request can all be approved & cancelled.
3. Verify ERC20 approve transaction confirmation header shows BraveWallet when created from native wallet, origin + favicon from dApp.
4. Verify creating & submitting/approving (test networks are fine) Solana & Ethereum transaction


## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
